### PR TITLE
Specify folder for git clone in `compile-pytorch-ipex.sh`; update IPEX pin

### LINKED ISSUE #2651: Cannot build pytorch with IPEX using `compile-pytorch-ipex.sh`
To reproduce: 
```
conda create --name triton-3.10 python=3.10   
conda activate triton-3.10
git clone https://github.com/intel/intel-xpu-backend-for-triton.git
cd /intel-xpu-backend-for-triton
./scripts/compile-triton.sh 
pip install matplotlib pandas tabulate
pip install wheel cmake  
./scripts/compile-pytorch-ipex.sh
```

This is the error:
```
....
Fetching submodule third_party/oneDNN
From https://github.com/oneapi-src/oneDNN
 * [new tag]             v3.4-rc    -> v3.4-rc
Fetching submodule third_party/xetla
fatal: remote error: upload-pack: not our ref 53a487515fa8d78a0df28b06d08c4049e26db194
fatal: remote error: upload-pack: not our ref 0debfce71cf3bc8b313aa2d6ad247f1a72adb6a0
fatal: remote error: upload-pack: not our ref ceaeb2556a40ed8f50fe0ab6e0d42f9cd9577786
Errors during submodule fetch:
        third_party/xetla
        third_party/oneDNN
        third_party/ideep
```

### PR DESCRIPTION
**Pull request by anmyachev**
Closes #2651

### CODE REVIEW

File: scripts/compile-pytorch-ipex.sh [hunk 1]
```diff
@@ -106,8 +106,7 @@ if [[ $BUILD_PYTORCH = true ]]; then
   rm -rf $PYTORCH_PROJ
 
   echo "**** Cloning $PYTORCH_PROJ ****"
-  cd $BASE
-  git clone --single-branch -b dev/triton-test-3.0 --recurse-submodules --jobs 8 https://github.com/Stonepia/pytorch.git
+  git clone --single-branch -b dev/triton-test-3.0 --recurse-submodules --jobs 8 https://github.com/Stonepia/pytorch.git $PYTORCH_PROJ
```
Comment (anmyachev) [inline scripts/compile-pytorch-ipex.sh:109]: By default was `pytorch`

**Review by whitneywhtsang — APPROVED** [review]

**Comment by whitneywhtsang** [comment]
> Closes #2651

The change LGTM, but it doesn't resolve the problem reported by #2651.

**Comment by anmyachev** [comment]
@Stonepia just to clarify, did you made force push to `dev/triton-test-3.0` branch? https://github.com/intel/intel-extension-for-pytorch/commit/15ef7db18b0a50101b41d9c78780d35ea7937ffc looks similar to the previous that we use.

**Comment by anmyachev** [comment]
@whitneywhtsang could you try with the last change?

**Comment by whitneywhtsang** [comment]
> @whitneywhtsang could you try with the last change?

It passes the reported problem, it is at the step `[ 74%] Built target dnnl_gpu_ocl` for a while, will reply again when it works e2e.

**Comment by whitneywhtsang** [comment]
```
[ 74%] Built target dnnl_gpu_jit
make: *** [Makefile:136: all] Error 2
Traceback (most recent call last):
  File "/home/jovyan/intel-xpu-backend-for-triton/.scripts_cache/intel-extension-for-pytorch/setup.py", line 1168, in <module>
    setup(
  File "/home/jovyan/.conda/envs/python-3.12/lib/python3.12/site-packages/setuptools/__init__.py", line 117, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by Stonepia** [comment]
Oh sorry for that. Yes we made a force push to the dev/triton-test-3.0 branch, because there is a token leak, that should be our fault. We will fix that.

**Comment by anmyachev** [comment]
Hi @Stonepia,

thanks for confirming! Don't worry, it's not that bad. I was able to build code with PTDB, even with python 3.12.



> ```
> [ 74%] Built target dnnl_gpu_jit
> make: *** [Makefile:136: all] Error 2
> Traceback (most recent call last):
>   File "/home/jovyan/intel-xpu-backend-for-triton/.scripts_cache/intel-extension-for-pytorch/setup.py", line 1168, in <module>
>     setup(
>   File "/home/jovyan/.conda/envs/python-3.12/lib/python3.12/site-packages/setuptools/__init__.py", line 117, in setup
>     return distutils.core.setup(**attrs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ```

@whitneywhtsang I can't reproduce that problem. If it's still there, I guess we can continue in a separate thread.

**Comment by whitneywhtsang** [comment]
> Well, this is kind of expected, our IPEX branch doesn’t involve working with 2025 compiler, or I missed something?

Make sense, verified it works for me with PTDB.

**Pull request merged**